packages/ilios-common/addon/components/course/rollover.js

```diff
@@ -31,27 +30,44 @@ export default class CourseRolloverComponent extends Component {
     for (let i = 0; i < 6; i++) {
       this.years.push(currentYear + i);
     }
+    this.title = this.args.course.title;
   }
 
-  load = restartableTask(async (event, [course]) => {
-    if (!course) {
-      return;
-    }
-    this.title = course.title;
-    const school = course.belongsTo('school').id();
-    this.academicYearCrossesCalendarYearBoundaries = await this.iliosConfig.itemFromConfig(
-      'academicYearCrossesCalendarYearBoundaries',
+  @cached
+  get academicYearCrossesCalendarYearBoundariesData() {
+    return new TrackedAsyncData(
+      this.iliosConfig.itemFromConfig('academicYearCrossesCalendarYearBoundaries'),
+    );
+  }
+
+  get academicYearCrossesCalendarYearBoundaries() {
+    return this.academicYearCrossesCalendarYearBoundariesData.isResolved
+      ? this.academicYearCrossesCalendarYearBoundariesData.value
+      : false;
+  }
+
+  @cached
+  get allCoursesData() {
```

Comment:
This is OK. It works. But it will force a reload of all courses (`store.query` is never cached).  My approach here would be to create a `get schoolData` that loads `this.args.course.school` and then have the `allCoursesData` getter to `this.school.courses`.

Reply:
I'm not exactly sure how to refactor this per your suggestion. Could you supply some code to help guide me? 😃

Reply:
Absolutely. Sort of free hand though so might have syntax errors:

Add a school data property:
```javascript
@cached
get schoolData() {
  return new TrackedAsyncData(this.args.course.school);
}
```

Then consume it to load all the courses like

```javascript
@cached
get coursesData() {
  return new TrackedAsyncData(this.schoolData.isResolved ? this.schoolData.value.courses : null);
}
```

Then use that to get the list of courses for the current school:

```javascript
get allCourses() {
if (!this.coursesData.isResolved || this.coursesData.value === null) {
return [];
}

return this.courses.data.value;
}
```

This will use ember-data's built in cache because all the courses in the school are already fetched from the API.